gen/class.go

```diff
@@ -126,38 +127,47 @@ func (g *generator) genClass(c *classData) error {
 		}
 	}
 
-	// let's make that constructor
 	g.bw.WriteRune('\n')
-	g.indent()
-	g.bw.WriteString("def __init__(self, ")
 
-	for n, i := range c.members {
-		if n < len(c.members)-1 {
-			g.bw.WriteString(fmt.Sprintf("%s: %s = None, ", i.Name(), i.Typename()))
-			continue
+	// let's make that constructor
+	if !c.noInit {
+		g.indent()
+		g.bw.WriteString("def __init__(self, ")
+
+		for n, i := range c.members {
+			if n < len(c.members)-1 {
+				g.bw.WriteString(fmt.Sprintf("%s: %s = None, ", i.Name(), i.Typename()))
+				continue
+			}
+
+			g.bw.WriteString(fmt.Sprintf("%s: %s = None", i.Name(), i.Typename()))
 		}
 
-		g.bw.WriteString(fmt.Sprintf("%s: %s = None", i.Name(), i.Typename()))
+		g.bw.WriteString(fmt.Sprintf(") -> %s: ...\n", c.name))
 	}
 
-	g.bw.WriteString(fmt.Sprintf(") -> %s: ...\n", c.name))
-
 	if c.hasName {
 		g.indent()
-		g.bw.WriteString(fmt.Sprintf("def Name(enumClass: %s) -> Any: ...\n", c.name))
+		g.bw.WriteString("@staticmethod\n")
+
+		g.indent()
+		g.bw.WriteString(fmt.Sprintf("def Name(number: %s) -> str: ...\n", c.name))
 	}
 
 	if c.hasValue {
 		g.indent()
-		g.bw.WriteString("def Value(memberName: str) -> Any: ...\n")
+		g.bw.WriteString("@staticmethod\n")
+
+		g.indent()
+		g.bw.WriteString(fmt.Sprintf("def Value(name: str) -> %s: ...\n", c.name))
 	}
 
 	for _, i := range c.types {
 		// enums need to be treated differently
 		if e, ok := i.(*protostub.Enum); ok {
 			for _, j := range e.Members {
 				g.indent()
-				g.bw.WriteString(fmt.Sprintf("%s: Any\n", j.Name()))
+				g.bw.WriteString(fmt.Sprintf("%s = cast(%s, %d)\n", j.Name(), c.name, j.Integer()))
```

Comment:
How we we access the integer position of each enum value?

Reply:
The members should be in `e.Members` in the correct order, so:

```
for n, j := range e.Members {
    // code
}
```

Where `n` is that position.

Reply:
The code is not necessarily that though. There's no reason why enums can't be specified as:

```
Foo = 4;
Bar = 123;
```

Reply:
That's fair enough. In this case it shouldn't matter, as the type stubs shouldn't affect anything to do with actual working code. Provided it gets the correct type, the value could be anything.

Reply:
Hmmm, OK. I see what you mean, but I also see how this could get very confusing. I'd rather we used the real integer values. Can you see of a way to extract them?

Reply:
Yeah I see what you mean! We can access the int from `proto.EnumField.Integer`, which can be found as a parameter to `VisitEnumField` in `enum.go`. Though, actually assigning a value to it shouldn't be necessary, we could just annotate the typename and leave it without a value.

Reply:
> Yeah I see what you mean! We can access the int from proto.EnumField.Integer, which can be found as a parameter to VisitEnumField in enum.go. 

Yes, I saw that, but there's no way to get that through to the code in class.go, without some changes.